File: 1693-sum-of-all-odd-length-subarrays/1693-sum-of-all-odd-length-subarrays.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int l = arr.length;
        int total = 0;
        for(int i = 0; i < l; i++){
            total += arr[i];
        }
        if(l % 2 == 1 && l != 1){
            total *= 2;
        }
        for(int windowSize = 3; windowSize <= l; windowSize += 2){
            if(windowSize == l && l % 2 == 1) {
                break;
            }
            int prefixSum = 0;
            for(int i = 0; i < windowSize; i++){
                prefixSum += arr[i];
            }
            total += prefixSum;
            for(int i = 1; i <= l - windowSize; i++){
                prefixSum += (arr[i + windowSize - 1] - arr[i - 1]);
                total += prefixSum;
            }
        }
        return total;
    }
}

File: 1693-sum-of-all-odd-length-subarrays/1693-sum-of-all-odd-length-subarrays.py
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        l = len(arr)
        total = sum(arr)
        for windowSize in range(3, l + 1, 2):
            prefixSum = 0
            for i in range(windowSize):
                prefixSum += arr[i]
            total += prefixSum
            for i in range(1, l - windowSize + 1):
                prefixSum += (arr[i + windowSize - 1] - arr[i - 1])
                total += prefixSum
        return total